com.arc.embeddedcdt/src/com/arc/embeddedcdt/dsf/utils/Configuration.java

```diff
@@ -48,10 +48,24 @@ private static boolean getAttribute(ILaunchConfiguration lc, String attribute, b
         }
     }
 
+    private static int getAttribute(ILaunchConfiguration lc, String attribute, int defaultValue) {
+        try {
+            return lc.getAttribute(attribute, defaultValue);
+        } catch (CoreException e) {
+            e.printStackTrace();
+            return defaultValue;
+        }
+    }
+
     public static String getProgramName(ILaunchConfiguration lc) {
         return getAttribute(lc, ICDTLaunchConfigurationConstants.ATTR_PROGRAM_NAME, "");
     }
 
+    public static int getDebuggerPluginVersion(ILaunchConfiguration lc){
+        return getAttribute(lc, LaunchConfigurationConstants.ATTR_DEBUGGER_PLUGIN_VERSION,
+                LaunchConfigurationConstants.UNREAL_DEBUGGER_PLUGIN_VERSION_NUMBER);
+    }
+
```

Comment:
Subsequently, you can move "checkPluginVersion" into a Configuration constructor - check would be done once, when configuration is created. So call of checkPluginVersion will be just one, it would be expected that all of the configuration users will use this Configuration class to ensure that check is done in all cases where needed.